src/librustc/lint/builtin.rs

```diff
@@ -149,6 +149,12 @@ declare_lint! {
     "safe access to extern statics was erroneously allowed"
 }
 
+declare_lint! {
+    pub TYPE_VS_VARIANT_AMBIGUITY,
+    Forbid,
+    "type vs variant ambiguity"
```

Comment:
Nit: maybe spell out "vs" as "versus" in both cases for slightly improved clarity. Most people will know what it means I think, but some people might thing it's some odd acronym relating to types?